lightningd/options.c

```diff
@@ -409,6 +409,9 @@ static void config_register_opts(struct lightningd *ld)
 			       opt_set_bool_arg, opt_show_bool,
 			       &ld->use_proxy_always, "Use the proxy always");
 
+	opt_register_noarg("--disable-dns", opt_set_bool, &ld->config.no_dns,
+			   "Disable DNS lookups of peers");
```

Comment:
I've been shying away from negative bools, so suggest we call it 'dns' and use opt_set_invbool here...